### PR TITLE
Use Autodiscovery instead

### PR DESCRIPTION
**Pull request by MarcelMue**
As discussed in https://github.com/giantswarm/giantswarm/issues/4543

### CODE REVIEW

**Review by xh3b4sd — APPROVED** [review]
That makes sense to me.

**Review by tuommaki — APPROVED** [review]
Yes! This auto-discovery seems a lot better than passing min/max/asg-name back and forth.

File: helm/kubernetes-cluster-autoscaler-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -41,7 +41,7 @@ spec:
             - --stderrthreshold=info
             - --cloud-provider=aws
             - --skip-nodes-with-local-storage=false
-            - --nodes={{ .Values.nodes.min }}:{{ .Values.nodes.max }}:{{ .Values.nodes.asg.name }}
+            - --nodes=-node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/{{ .Values.cluster.id }}
```
[the remark this answers is not in the document]
Comment (teemow) [reply helm/kubernetes-cluster-autoscaler-chart/templates/deployment.yaml]: From the docs:

> `--node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/<YOUR CLUSTER NAME>` will find the ASGs where those tag keys exist. It does not matter what value the tags have.


https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/aws#auto-discovery-setup

[the remark this answers is not in the document]
Comment (tuommaki) [reply helm/kubernetes-cluster-autoscaler-chart/templates/deployment.yaml]: Ah, but of course. Stupid me. Adding this to `aws-operator` now. :+1:

**Pull request merged**